secboot/secboot_hooks.go

```diff
@@ -135,30 +142,45 @@ func ResealKeysWithFDESetupHook(keyFiles []string, primaryKeyFile string, models
 	for _, model := range models {
 		sbModels = append(sbModels, model)
 	}
-	for _, keyFile := range keyFiles {
-		loadedKey := &defaultKeyLoader{}
-		const hintExpectFDEHook = true
-		if err := readKeyFile(keyFile, loadedKey, hintExpectFDEHook); err != nil {
-			return err
-		}
+	for _, key := range keys {
+		var keyDataWriter sb.KeyDataWriter
 
-		// Non-nil FDEHookKeyV1 indicates that V1 hook key is used
-		if loadedKey.FDEHookKeyV1 != nil {
-			// V1 keys do not need resealing
-			continue
-		}
+		keyData, tokenErr := readKeyToken(key.DevicePath, key.SlotName)
```

Comment:
isn't some of this code very similar to readKeyData, could that be reused? could it become a method on KeyDataLocation ?

Reply:
I can try. The issue here is the v1 format for hooks.

Reply:
I have extracted the token part. But reading from file differs because of context.